qiskit/circuit/instruction.py

```diff
@@ -136,6 +136,10 @@ def params(self):
     @params.setter
     def params(self, parameters):
         self._params = []
+
+        if isinstance(parameters, numpy.ndarray):
+            parameters = parameters.tolist()
+
         for single_param in parameters:
             # example: u2(pi/2, sin(pi/4))
             if isinstance(single_param, (ParameterExpression)):
```

Comment:
The condition `elif isinstance(single_param, numpy.number):` should also be moved to be checked before the `isinstance` checks for Python numeric types `int, float, complex` since Numpy types pass that check, but are infact different types.

Reply:
they all have the same effect, right? the order should not affect.

Reply:
No it doesn't and this is the whole reason for the bug.

if you pass in `params = [np.float64(1.0)]`. It will evaluate `isinstance(single_param, float)` to `True` and append the Numpy type before checking `isinstance(single_param, np.number)` which would convert to a native python type before appending using `single_param.item()`. Similarly for numpy complex and int types.

Reply:
but there is no conversion currently... or am I missing something? In any case there is a `                self._params.append(single_param)`